rest-api-spec/src/main/resources/rest-api-spec/api/cluster.create_data_stream.json

```diff
@@ -0,0 +1,31 @@
+{
+  "cluster.create_data_stream":{
```

Comment:
After chatting with Henning, I also came to the conclusion that data stream apis should be concidere indices based operations. (Ideally we should have a data_stream notion, but we need to think more about this and how this would work in security). So I will revert the commit in this pr that changes data stream crud apis back to indices based apis.

Reply:
the component template apis are cluster operations, I suspect those may need to be changed also, can you explain the reasoning and how it will relate to security?

Reply:
Just like indices based apis, data stream apis targets a namespace. So someone may be allowed to create a data stream for logs-* but not for events-*.

Ideally there should be a data stream privilege that handles this correctly, because data streams aren't indices and a data stream may have indices that don't share the data stream name as common prefix. But for now let's stick with `indices:` based action names and group the apis in the indices client, until we a better there is a better understanding how security and data streams should integrate.

I think component templates should remain cluster based apis. The reason is that these resources don't apply to a namespace. The index template (based on the specified pattern) that use component templates should be an index based action/operation, since they apply the an index namespace and soon also to a data stream namespace. 

However currently index templates are treated as cluster privilege (see ClusterPrivilegeResolver line 196 in master), even though the action names start with `indices:` prefix. With data streams we should properly fix this.

Reply:
Okay, I originally had component template APIs using `indices:` to match the existing template stuff, but I did end up changing it because that requires the request to provide the indices that it is going to apply to, and component templates don't apply to indices. +1 to properly fix the template exception with data streams.